src/util/map/SetMultiMap.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * A SetMultiMap is a Map where a single key can have multiple unique values.
+ * Deleting a key removes all bindings with this key from the Map.
+ * Setting a value for a key replaces all previous bindings with this key.
+ * Using an empty Set when calling the `set` function is the same as deleting that key.
+ */
+export interface SetMultiMap<TKey, TVal> extends Map<TKey, TVal | ReadonlySet<TVal>> {
```

Comment:
```suggestion
export interface MultiMap<TKey, TVal> extends Map<TKey, TVal | ReadonlySet<TVal>> {
```
no? 😇

Reply:
As mentioned in the previous discussion, expected behaviour of a `MultiMap` allows multiple values for a single key, while a `SetMultiMap` only allows unique values. So I want to be consistent with that.

Reply:
But then then the "multi" is superfluous.
`SetMap`? 😅 Ah it's okay.